bin/cron/pardot_sync

```diff
@@ -0,0 +1,419 @@
+  log "   Completed in " + ((Time.now) - start).to_s + " seconds"
+
+  start = Time.now
+  log "Updating Professional Learning Partner role"
+  append_plp_to_role_list
+  log "   Completed in " + ((Time.now) - start).to_s + " seconds"
+end
+
+def append_to_role_list_from_permission(permission_name, dest_value)
+  PEGASUS_DB.run "
+    UPDATE contacts_rollup
+	    INNER JOIN #{DASHBOARD_DB_NAME}.users AS users ON users.id = contacts_rollup.dashboard_user_id
+	    INNER JOIN #{DASHBOARD_DB_NAME}.user_permissions AS user_permissions ON user_permissions.user_id = users.id
+	  SET roles = CONCAT(COALESCE(CONCAT(roles,','),''),'#{dest_value}')
+    WHERE LENGTH(users.email) > 0
+		AND user_permissions.permission = '#{permission_name}' AND contacts_rollup.id > 0"
+end
+
+def append_to_list_field_from_form(form_kinds, dest_field, dest_value)
+  start = Time.now
+  log "Appending '#{dest_field}' field with #{dest_value} from forms of kind #{form_kinds}"
+
+  PEGASUS_DB.run "
+    UPDATE contacts_rollup
+      INNER JOIN forms ON forms.email = contacts_rollup.email
+	  SET #{dest_field} = CONCAT(COALESCE(CONCAT(#{dest_field},','),''),#{dest_value})
+    WHERE forms.kind IN (#{form_kinds}) AND contacts_rollup.id > 0"
+  log "   Completed in " + ((Time.now) - start).to_s + " seconds"
+end
+
+def append_plp_to_role_list
+  PEGASUS_DB.run "
+    UPDATE contacts_rollup
+	    INNER JOIN #{DASHBOARD_DB_NAME}.users AS users ON users.id = contacts_rollup.dashboard_user_id
+	    INNER JOIN #{DASHBOARD_DB_NAME}.professional_learning_partners AS professional_learning_partners ON professional_learning_partners.contact_id = users.id
+	  SET roles = CONCAT(COALESCE(CONCAT(roles,','),''),'PLP')
+    WHERE LENGTH(users.email) > 0 AND contacts_rollup.id > 0"
+end
+
+# List of valid PD courses. If more courses get added, need to update this list and also schema in Pardot. We
+# need to filter here to known courses in the Pardot schema - we can't blindly pass values through
+COURSE_LIST = "'CS Fundamentals','CS in Algebra','CS in Science','CS Principles','Exploring Computer Science'"
+
+def update_facilitator_types
+  start = Time.now
+  log "Updating facilitator types"
+  PEGASUS_DB.run "
+    UPDATE contacts_rollup,
+    (SELECT facilitator_id, group_concat(course) AS courses FROM
+      (SELECT DISTINCT facilitator_id,course FROM #{DASHBOARD_DB_NAME}.pd_course_facilitators
+	      WHERE COURSE IN (#{COURSE_LIST})
+	      ORDER by 1,2
+      ) q
+      GROUP BY facilitator_id
+    ) src
+	  SET contacts_rollup.facilitator_type = src.courses
+    WHERE contacts_rollup.dashboard_user_id = src.facilitator_id"
+  log "   Completed in " + ((Time.now) - start).to_s + " seconds"
+end
+
+def update_professional_learning_enrollment
+  start = Time.now
+  log "Updating professional learning enrollment"
+  PEGASUS_DB.run "
+    UPDATE contacts_rollup,
+      (SELECT user_id, group_concat(course) AS courses FROM
```

Comment:
style: I think convention is for [GROUP_CONCAT](http://dev.mysql.com/doc/refman/5.7/en/group-by-functions.html#function_group-concat) to be capitalized.